chris/single-number.go

```diff
@@ -0,0 +1,23 @@
+package chris
+
+func singleNumber(nums []int) int {
+
+	if len(nums) == 1 {
+		return nums[0]
+	}
+
+	result := map[int]int{}
+	for _, v := range nums {
```

Comment:
In Go, how does this syntax work? Is it basically `for "Index in the collection", "value itself" in "collection"`?

Reply:
Basically yes, and you use _ to ignore that value. Range will give me both and I don't actually care about the index so I just ignore that.

Reply:
Neat. I haven't had a chance to goof around with Go very much, so this is a neat way to start looking at it - comparing solutions to a common problem.

Reply:
Yeah, I need to look at your TS solution to get a better idea of how that works too.